src/main/resources/script.sh

```diff
@@ -0,0 +1,1241 @@
+chmod 755 *
+
+# Record the timestamp after downloads
+AFTERDOWNLOAD=$(date +%s)
+
+# Stop log for inputs download
+stopLog inputs_download
+
+
+startLog application_environment
+echo "$variables"
+
+# Iterate through each variable in $variables and export them
+# Assuming $variables is a string containing key-value pairs separated by space
+for variable in $variables; do
+    key=$(echo "$variable" | cut -d'=' -f1)
+    value=$(echo "$variable" | cut -d'=' -f2)
+    export "$key"="$value"
+done
+
+# Stop log for application environment
+stopLog application_environment
+
+startLog application_execution
+
+# Perform service call if minor status is enabled
+if [[ $minorStatusEnabled == true && $serviceCall ]]; then
+    $serviceCall ${MOTEUR_WORKFLOWID} ${JOBID} 4
+fi
+
+# Add a delay to ensure file creation before proceeding
+echo "BEFORE_EXECUTION_REFERENCE" > BEFORE_EXECUTION_REFERENCE_FILE
+sleep 1
+
+checkBosh $BOSH_CVMFS_PATH
+
+####################################################################################################
+# Clone udocker (A basic user tool to execute simple docker containers in batch or interactive systems without root privileges)
+if ! command -v docker
+then
+    download_udocker
+fi
+####################################################################################################
+
+# Export current directory to LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${PWD}:${LD_LIBRARY_PATH}
+
+# Execute the command
+PYTHONPATH=".:$PYTHONPATH" $BOSHEXEC exec launch ../$workflowFile ../inv/$invocationJson -v $PWD/../cache:$PWD/../cache
+
+# Check if execution was successful
+if [ $? -ne 0 ]; then
+    error "Exiting with return value 6"
+    BEFOREUPLOAD=$(date +%s)
+    info "Execution time: $(expr ${BEFOREUPLOAD} - ${AFTERDOWNLOAD}) seconds"
+    stopLog application_execution
+    cleanup
+    exit 6
+fi
+
+BEFOREUPLOAD=$(date +%s)
+stopLog application_execution
+
+info "Execution time was $(expr ${BEFOREUPLOAD} - ${AFTERDOWNLOAD})s"
+
+
+####################################################################################################
+
+provenanceFile="$BASEDIR/$DIRNAME.sh.provenance.json"
+echo $provenanceFile THIS IS FOR DEBUG and TESTs ONLY
+copyProvenanceFile "$provenanceFile"
+
+startLog results_upload
+
+# Perform service call if minor status is enabled
+if [[ $minorStatusEnabled == true && $serviceCall ]]; then
+    $serviceCall ${MOTEUR_WORKFLOWID} ${JOBID} 5
+fi
+
+
+# Extract the file names and store them in a bash array (first method is commented out since jq has imcomplete support in some linux distributions)
+file_names=($(jq -r '.["public-output"]["output-files"] | to_entries[] | .value["file-name"]' "$provenanceFile")) 
+#file_names=($(grep -o '"file-name": *"[^"]*"' "$provenanceFile" | awk -F': ' '{print $2}' | tr -d '"' | tr '\n' ' ')) //experimental
+
+# Remove square brackets from uploadURI (we assume UploadURI will always be a single string)
+uploadURI=$(echo "$uploadURI" | sed 's/^\[//; s/\]$//')
```

Comment:
Did you have an issue with brackets ?

Reply:
uploadURI is a List of URIs. We either need to change it to a URI (or String) in java, or write a logic in the bash to handle List of URIs. Then the bracket won't be an issue.